pip/req/req_install.py

```diff
@@ -805,6 +805,19 @@ def _clean_zip_name(self, name, prefix):
         name = name.replace(os.path.sep, '/')
         return name
 
+    def _get_setup_py_command(self, setup_py_path):
+        cmd = (
+            'import setuptools, tokenize ;'
+            '__file__ = %r ;'
+            '_open = getattr(tokenize, "open", open) ;'
+            'setup_py = _open(__file__).read() ;'
+            'setup_py = setup_py.replace("\\r\\n", "\\n") ;'
+            'codeobj = compile(setup_py, __file__, "exec") ;'
+            'exec(codeobj)'
+        )
```

Comment:
You've restructured the code here, and introduced a number of local variables, which will be visible within the exec. I can't imagine how those extra variables could alter the behaviour of the setup.py invocation, but it is a change in behaviour. I'm inclined to say to leave the injection code as it was - the readability benefit seems minimal.